paging-runtime-composeui/src/commonMain/kotlin/app/cash/paging/LazyPagingItems.kt

```diff
@@ -0,0 +1,320 @@
+       *
+       * Checks for null LOGGER because other runtime
+       * entry points to paging can also inject a Logger
+       */
+      LOGGER = LOGGER ?: createLogger()
+    }
+  }
+}
+
+private val IncompleteLoadState = LoadState.NotLoading(false)
+private val InitialLoadStates = LoadStates(
+  LoadState.Loading,
+  IncompleteLoadState,
+  IncompleteLoadState
+)
+
+
+/**
+ * Collects values from this [Flow] of [PagingData] and represents them inside a [LazyPagingItems]
+ * instance. The [LazyPagingItems] instance can be used by the [items] and [itemsIndexed] methods
+ * from [LazyListScope] in order to display the data obtained from a [Flow] of [PagingData].
+ *
+ * @param context the [CoroutineContext] to perform the collection of [PagingData]
+ * and [CombinedLoadStates].
+ */
+@Composable
+fun <T : Any> Flow<PagingData<T>>.collectAsLazyPagingItems(
+  context: CoroutineContext = EmptyCoroutineContext
+): LazyPagingItems<T> {
+
+  val lazyPagingItems = remember(this) { LazyPagingItems(this) }
+
+  LaunchedEffect(lazyPagingItems) {
+    if (context == EmptyCoroutineContext) {
+      lazyPagingItems.collectPagingData()
+    } else {
+      withContext(context) {
+        lazyPagingItems.collectPagingData()
+      }
+    }
+  }
+
+  LaunchedEffect(lazyPagingItems) {
+    if (context == EmptyCoroutineContext) {
+      lazyPagingItems.collectLoadState()
+    } else {
+      withContext(context) {
+        lazyPagingItems.collectLoadState()
+      }
+    }
+  }
+
+  return lazyPagingItems
+}
+
+/**
+ * Adds the [LazyPagingItems] and their content to the scope. The range from 0 (inclusive) to
+ * [LazyPagingItems.itemCount] (exclusive) always represents the full range of presentable items,
+ * because every event from [PagingDataDiffer] will trigger a recomposition.
+ *
+ * @param items the items received from a [Flow] of [PagingData].
+ * @param key a factory of stable and unique keys representing the item. Using the same key
+ * for multiple items in the list is not allowed. Type of the key should be saveable
+ * via Bundle on Android. If null is passed the position in the list will represent the key.
+ * When you specify the key the scroll position will be maintained based on the key, which
+ * means if you add/remove items before the current visible item the item with the given key
+ * will be kept as the first visible one.
+ * @param itemContent the content displayed by a single item. In case the item is `null`, the
+ * [itemContent] method should handle the logic of displaying a placeholder instead of the main
+ * content displayed by an item which is not `null`.
+ */
+fun <T : Any> LazyListScope.items(
+  items: LazyPagingItems<T>,
+  key: ((item: T) -> Any)? = null,
+  itemContent: @Composable LazyItemScope.(value: T?) -> Unit
+) {
+  items(
+    count = items.itemCount,
+    key = if (key == null) null else { index ->
+      val item = items.peek(index)
+      if (item == null) {
+        createPlaceholderKey(index)
```

Comment:
Here's where I extracted out the android Parceler to target specific implementations